cmd/empire/factories.go

```diff
@@ -269,7 +269,9 @@ func newHBReporter(key, env string) (reporter.Reporter, error) {
 
 func newAuthProvider(c *cli.Context) (dockerauth.AuthProvider, error) {
 	provider := dockerauth.NewMultiAuthProvider()
-	provider.AddProvider(dockerauth.NewECRAuthProvider(ecr.New(newConfigProvider(c))))
+	provider.AddProvider(dockerauth.NewECRAuthProvider(func(region string) dockerauth.ECR {
+		return ecr.New(newConfigProvider(c), &aws.Config{Region: aws.String(region)})
```

Comment:
Perhaps it's smarter to close over an instance of ConfigProvider, rather than construct it every time the factory is called. Looks like the AWS SDK will [create a copy](https://github.com/aws/aws-sdk-go/blob/af3026711cd873ccb4e87a7142fa8ca0be8f2aef/aws/session/session.go#L108) of the session, which should make it safe to do so.